### PR TITLE
feat: added recoverytool tests

### PR DESCRIPTION
**Pull request by douglaz**
Added tests for `utxos`and `epochs` method. Reviewed the README

### CODE REVIEW

**Comment by douglaz** [comment]
@elsirion Not sure the best way to test the `direct` method. How is one supposed to get the tweaks?

**Review by elsirion — APPROVED** [review]
Thx a lot! One thing we might want to add in the future is doing a peg-out before, so we can test that change also gets recovered :)

**Comment by justinmoon** [comment]
Very nice

File: recoverytool/src/main.rs [hunk 1]
```diff
@@ -67,6 +67,10 @@ struct RecoveryTool {
     /// Network to operate on, has to be specified if --cfg isn't present
     #[arg(long, default_value = "bitcoin", requires = "descriptor")]
     network: Network,
+    /// Open the database in read-only mode, useful for debugging, should not be
+    /// used in production
+    #[arg(long)]
+    readonly: bool,
```
Comment (bradleystachurski) [inline recoverytool/src/main.rs:73]: Useful 👍

**Pull request merged**